### PR TITLE
Ticket 7

### PR DESCRIPTION
**Pull request by 1991hannah**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -162,7 +162,83 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        const data = {
+            username: 123,
+            body: "Great article"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("One or more inputs incorrectly formatted in request")
```
Comment (mezzDavies) [inline __tests__/app.test.js]: A side note!

If you are fed up of making new messages it is perfectly fine to simply respond with "bad request" for all these bad request errors. We don't have to be _that_ helpful or specific as there will be/is documentation to show how to use the api. Sending "bad request" is putting the ball in the court of the client to use the api properly and means as a dev we don't have to cover each error with unique messages! (Also it is possible to give information about our database if we get too specific which we don't want to do)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -162,7 +162,83 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        const data = {
+            body: "This is a great article",
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.comment).toMatchObject({
```
Comment (mezzDavies) [inline __tests__/app.test.js:181]: We could assert the comment has the values we sent, present on the correct properties here.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -162,7 +162,83 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+            expect(body.msg).toBe("Request is missing key information")
+        })
+    })
+    test("status 400: valid article id but no corresponding article", () => {
+        const data = {
+            body: "Wow fantastic",
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/9999/comments')
```
Comment (mezzDavies) [inline __tests__/app.test.js:234]: Is this message - "one or more inputs incorrectly formatted in request" true for this test? the id is of the correct type - a number - and the comment is correctly formatted too. Therefore is 400 the appropriate status? 

As the article_id is valid just is _not found_ and the form of the comment being sent is correct there is a more appropriate error status and message, I reckon. 🙂

**Review by mezzDavies — CHANGES_REQUESTED** [review]
This is looking good on the whole 👍 

I have left a couple of comments to think about/refactor and there is another test for you to think about...

- what if the username is provided but doesn't exist in the database 🤔 

Please have a go at this and then resubmit

File: __tests__/app.test.js (hunk 3 above)
Comment (1991hannah) [reply __tests__/app.test.js:234]: Hey Mezz! I thought it was supposed to be a 404 but when I ran the test the error came back as the same PSQL error (23503) which I got when inputting the incorrect information which I'd already assigned the "400: one or more inputs incorrectly formatted in request" error to - is there a way to assign two different responses if the PSQL error is the same?

File: __tests__/app.test.js (hunk 1 above)
Comment (1991hannah) [reply __tests__/app.test.js]: That's good to know!

File: __tests__/app.test.js [hunk 4]
```diff
@@ -153,16 +153,94 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+            expect(body.comment.body).toEqual(data.body)
+            expect(body.comment.author).toEqual(data.username)
+            expect(body.comment).toMatchObject({
+                article_id:expect.any(Number),
+                author:expect.any(String),
+                body:expect.any(String),
+                comment_id:expect.any(Number),
+                created_at:expect.any(String),
+                votes:expect.any(Number)
+            })
```
Comment (Keb84) [inline __tests__/app.test.js:188]: You could just assert that you expect the comment to have the values that you sent and they are on the correct properties - just something to consider 💭

**Review by Keb84 — APPROVED** [review]
nice work on the changes for this ticket 🤸 merge when ready 🚀

**Pull request merged**